### PR TITLE
Migrate CI to Github Actions

### PR DESCRIPTION
**Pull request by allanlewis**
Following https://github.com/ataylor32/django-username-email/pull/23#issuecomment-1151827952, this PR migrates CI from Travis to Github Actions.

See allanlewis/django-username-email#2 for CI results as the workflows don't exist in this repo.

### CODE REVIEW

File: .github/workflows/python-publish.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+          pip install build==0.8.*
+
+      - name: Build package
+        run: python -m build
+
+      - name: Publish package
+        uses: pypa/gh-action-pypi-publish@v1
+        with:
+          user: __token__
+          password: ${{ secrets.PYPI_API_TOKEN }}
```
Comment (allanlewis) [inline .github/workflows/python-publish.yml:42]: @ataylor32 you'll need to add a secret for your API token.

Comment (ataylor32) [reply .github/workflows/python-publish.yml:42]: Done

**Comment by ataylor32** [comment]
Is there anything else that needs to be done before this can be merged?

**Comment by allanlewis** [comment]
> Is there anything else that needs to be done before this can be merged?

It just needs testing. I'll try to find time to do that on my fork. I won't be able to test the deployment workflow as I don't have a PyPI account but I take it the action works, and I've configured it based on Github's template.

**Comment by allanlewis** [comment]
I think this should be working now. As I said, I can't help much with the package-publishing side, I think your best option is to merge and test that yourself, maybe using a test account. The [docs for the action I used][1] should help if it doesn't work.

[1]: https://github.com/marketplace/actions/pypi-publish

**Comment by ataylor32** [comment]
Once I look into what changes (if any) are needed for Django 4.1, I'll test out the actions you made. I'll merge them for now and update them if needed. I really appreciate your time on this. Thank you!

**Comment by allanlewis** [comment]
Glad I could help 🙂

**Pull request merged**